### PR TITLE
Location Onboarding

### PR DESCRIPTION
**Pull request by DarkMio**
tldr: 

- asking for location permission
- handling error cases
- urging to start a certain start location
- introduces floating-ui for the selection behaviour
- localizable texts are in a separate file
- use the location result after onboarding has been finished
  - (that is relatively poor, maybe there's a better way, because currently it's using `useEffect`)

https://github.com/user-attachments/assets/b7f206f2-c69b-4d1d-9381-684e5532d62c

### CODE REVIEW

File: src/components/MapNew/MapView.tsx [hunk 1]
```diff
@@ -74,15 +74,31 @@ export default function MapView(props: IProps) {
...
     longitude: (longitude && parseFloat(longitude)) || 13.3,
     zoom: (zoom && parseFloat(zoom)) || (latitude && longitude ? 18 : 10),
+    lastUpdate: 0,
   })
 
+  React.useEffect(() => {
+    if (!latitude || !longitude) {
+      return
+    }
+    // poor but effective: the map allows location jumping from URI, if the map hasn't been touched for some time
```
Comment (opyh) [inline src/components/MapNew/MapView.tsx:91]: 😅 Okay for now

File: src/components/Onboarding/LocationFailedStep.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+const Container = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: 24px;
+
+  .title {
+    @media (min-width: 414px) {
+      font-size: 1.25rem;
+    }
+    @media (min-height: 414px) {
```
Comment (opyh) [inline src/components/Onboarding/LocationFailedStep.tsx:21]: Duplicate

File: src/components/Onboarding/LocationNoPermissionStep.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+import { KomootPhotonResultFeature } from '../../lib/fetchers/fetchPlacesOnKomootPhoton'
+
+const Container = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  gap: 24px;
+
+  h1 {
+    @media (min-width: 414px) {
```
Comment (opyh) [inline src/components/Onboarding/LocationNoPermissionStep.tsx:19]: Duplicate

File: src/components/Onboarding/LocationStep.tsx [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+
+      onGeneralError(error)
+    })
+  }, [onAccept, stage, setStage, onGeneralError, maxRetries, onFailed])
+
+  const isAcquiring = stage.stage === 'acquiring'
+
+  // when retries fail, show an additional hint
+  const primaryText = `${LocationStepPrimaryText}${
+    stage.retries > 0 ? `\n\n${LocationStepAdditionalHint('about:blank')}` : ''
```
Comment (opyh) [inline src/components/Onboarding/LocationStep.tsx:157]: There is a helpful `goToLocationSettings()` function in `goToLocationSettings.ts` :)

File: src/components/shared/ModalDialog.tsx [hunk 5]
```diff
@@ -16,8 +16,8 @@ type Props = {
 };
 
 const StyledSection = styled.section`
-  z-index: 10000000000;
-
+  z-index: 10000;
```
Comment (opyh) [inline src/components/shared/ModalDialog.tsx:19]: 😇

File: src/lib/fetchers/fetchPlacesOnKomootPhoton.ts [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+            : stringOrStringArray.map((value) => [key, value])
+        ),
+      ),
+  ])
+
+  if (supportedLanguageCodes.includes(languageCode)) {
+    queryParameters.append('lang', languageCode)
+  }
+
+  const url = `https://photon.komoot.io/api/?${queryParameters.toString()}`
```
Comment (opyh) [inline src/lib/fetchers/fetchPlacesOnKomootPhoton.ts:57]: When entering "Berlin", does it find the most relevant Berlin? We could bias the result using the location from the country we get from the CDN (via IP).

**Review by opyh — CHANGES_REQUESTED** [review]
Only some small nitpick requests. Nice and sooo fast!

File: src/lib/fetchers/fetchPlacesOnKomootPhoton.ts (hunk 6 above)
Comment (DarkMio) [reply src/lib/fetchers/fetchPlacesOnKomootPhoton.ts:57]: I'm not certain how to do it and can't find useful references how it's being used on other components.

**Comment by DarkMio** [comment]
@opyh location bias is in, merge when approved

**Comment by opyh** [comment]
Thanks for all the nice cleanups! :D

**Pull request merged**